sympy/integrals/manualintegrate.py

```diff
@@ -342,7 +342,10 @@ def _parts_rule(integrand, symbol):
     # log, inverse trig, algebraic (polynomial), trigonometric, exponential
     def pull_out_polys(integrand):
         integrand = integrand.together()
-        polys = [arg for arg in integrand.args if arg.is_polynomial(symbol)]
+        polys = [arg for arg in integrand.args if (arg.is_polynomial(symbol) or \
+                                                   (isinstance(arg, sympy.Pow) and \
```

Comment:
What is the purpose of these additions? According to the name this method aims to extract polynomials. But these additions would include powers with noninteger and negative exponents.